src/typesystem.rs

```diff
@@ -208,7 +211,7 @@ impl DataTypeInformation {
         match self {
```

Comment:
How about we add an "&Index" to the get size? I prefer that to the unimplemented string size. It would also let us calculate the size for the rest of the types.

Reply:
since half of the variants are note implemented it tells me that we basically only use this method in very very rare situations ... 
maybe we can rethink the whole type-calculation thing? I has no relevance for this commit